packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -104,9 +108,27 @@ class DatePickerInput extends React.Component {
   constructor(props) {
     super(props);
 
+    this.state = {};
+
+    if (this.props.value) {
+      this.state = {
+        day: this.props.value.substr(3, 2),
+        month: this.props.value.substr(0, 2),
```

Comment:
We can't guarantee that the first two digits are the month due to the different formats.

Reply:
Updated: [452f10b](https://github.com/cerner/terra-framework/pull/1004/commits/452f10bd2fac12ea3c50aff9d7b90d0cff1d3b01). Since the render changes based on the format, I formatted whatever the initial input value is to something consistent.

Reply:
Made some updated to this logic that... actually make sense.